### PR TITLE
[DSPDC-1699] Rename HCA load pipeline and add stub of datafile load branch

### PR DESCRIPTION
**Pull request by aherbst-broad**
## Why

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1699)
We need to implement the "load data files" branch of the HCA load pipeline in Dagster. 
This is the first of multiple PRs to implement this functionality. 

## This PR
* Renames the `stage_data` pipeline to `load_hca` and updates references
* Creates a `load_data_files` solid that is the stub of the branch to be implemented

### CODE REVIEW

File: orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/stage_data.py [hunk 1]
```diff
@@ -5,6 +5,10 @@
 from google.cloud.bigquery import Dataset
 
 
+class HcaStagingDatasetName(str):
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/stage_data.py:8]: The load data files pipeline will be passing many `str` parameters around, I am using this to bring some semblance of type safety to those parameters. Not 100% sure on the placement in this file, if folks have opinions.

Comment (ThePletch) [reply orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/stage_data.py:8]: i think custom types fit best in `hca_orchestration/support/typing.py`, with the potential to split that into subfiles in the `typing` directory instead if that file becomes too large or too disparate in function

File: orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/stage_data.py [hunk 2]
```diff
@@ -60,9 +64,9 @@ def pre_process_metadata(context: AbstractComputeExecutionContext) -> Nothing:
         "staging_table_expiration_ms": Int
     },
     input_defs=[InputDefinition("start", Nothing)],
-    output_defs=[OutputDefinition(name="staging_dataset_name", dagster_type=String)]
+    output_defs=[OutputDefinition(name="staging_dataset_name", dagster_type=HcaStagingDatasetName)]
```
Comment (ThePletch) [inline orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/stage_data.py:67]: output definition can be omitted for solids that return a single output - its type is inferred from the annotated return type

File: orchestration/dagster_orchestration/hca_orchestration/pipelines/load_hca.py [hunk 3]
```diff
@@ -61,5 +62,6 @@
 @pipeline(
     mode_defs=[prod_mode, dev_mode, local_mode, test_mode]
 )
-def stage_data() -> None:
-    create_staging_dataset(pre_process_metadata(clear_staging_dir()))
+def load_hca() -> None:
+    staging_dataset: HcaStagingDatasetName = create_staging_dataset(pre_process_metadata(clear_staging_dir()))
```
Comment (ThePletch) [inline orchestration/dagster_orchestration/hca_orchestration/pipelines/load_hca.py:66]: this type annotation should be redundant - it'll be enforced by the input schema for `import_data_files`

**Review by ThePletch — APPROVED** [review]
a few comments on code organization and dagster annotations, but the general pattern here looks good

**Review by raaidbroad — APPROVED** [review]
per usual not much more to add than steve's comments!

File: orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/stage_data.py (hunk 2 above)
Comment (aherbst-broad) [reply orchestration/dagster_orchestration/hca_orchestration/solids/load_hca/stage_data.py]: 👍 Interestingly, the output gets named "result" when pulling it via output_value() in tests.

**Pull request merged**